WoofWare.Expect.Test/TestSnapshotFinding.fs

```diff
@@ -211,3 +220,724 @@ more""
+        expect {
+            snapshot """"""""""""
+            return """"
+        }
+
+    let onlyWhitespace () =
+        expect {
+            snapshot ""   \t\n   ""
+            return ""whitespace""
+        }
+
+    let quotesInQuotes () =
+        expect {
+            snapshot @""
+
+Just newlines!
+
+
+""
+            return ""quotes""
+        }
+
+    let tripleQuotesInContent () =
+        expect {
+            snapshot """"""This has """""""""""" in the middle""""""
+            return ""triple""
+        }
+
+    let backslashesGalore () =
+        expect {
+            snapshot @""C:\Users\Test\Documents\file.txt""
+            return ""path""
+        }
+
+    let veryLongLine () =
+        expect {
+            snapshot @""This is a very long line that goes on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and on and contains over 300 characters to test how the parser handles very long single-line strings""
+            return ""long line""
+        }
+
+    let leadingNewlines () =
+        expect {
+            snapshot """"""
+
+Starts with newlines""""""
+            return ""leading""
+        }
+
+    let trailingNewlines () =
+        expect {
+            snapshot """"""Ends with newlines
+
+
+""""""
+            return ""trailing""
+        }
+"
+
+            return
+                SnapshotUpdate.updateSnapshotAtLine source 28 "\n\nJust newlines!\n\n\n"
+                |> String.concat "\n"
+        }
+
+    [<Test>]
+    let ``Complex nested JSON with Unicode`` () =
+        let source =
+            Assembly.getEmbeddedResource typeof<Dummy>.Assembly "MultilineComplex.fs"
+            |> _.Split('\n')
+
+        expect {
+            snapshot
+                @"namespace BigExample
+
+open WoofWare.Expect
+
+module MyModule =
+    let veryLongMultiline () =
+        expect {
+            snapshot
+                """"""Line 1
+Line 2
+Line 3
+Line 4
+Line 5
+Line 6
+Line 7
+Line 8
+Line 9
+Line 10
+    Indented line 11
+        More indented line 12
+Line 13
+Line 14
+Line 15""""""
+
+            return ""long""
+        }
+
+    let multilineWithEmptyLines () =
+        expect {
+            snapshot
+                @""First line
+
+Third line
+
+
+Sixth line""
+
+            return ""empty lines""
+        }
+
+    let multilineWithSpecialChars () =
+        expect {
+            snapshot
+                """"""Special chars:
+    Tab:	here
+    Quotes: ""double"" and 'single'
+    Backslash: \ and \\
+    Unicode: 🎯
+    Regex: .*+?[]""""""
+
+            return ""special""
+        }
+
+    let multilineJson () =
+        expect {
+            snapshotJson
+                @""{
+  """"name"""": """"test with émojis 🎯"""",
```

Comment:
Double escape of `"` here: once for the fact that the test string is multi-line, and once for the fact that it contains a quoted multiline string.